.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,132 @@
+name: Emscripten
+
+on: [push, pull_request]
+
+jobs:
+  build-win10:
+    strategy:
+      matrix:
+        config: [Debug, Release]
+
+    runs-on: windows-latest
+    name: Emscripten -> Windows 10, ${{ matrix.config }}
+
+    steps:
+    - name: Clone repository
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+
+    - name: Format validation
+      shell: cmd
+      run: |
+        cd BuildTools\FormatValidation
+        validate_format_win.bat
+
+    - name: Download Ninja
+      uses: actions/checkout@master
+
+    - name: Configure Ninja
+      uses: seanmiddleditch/gha-setup-ninja@master
+
+    - name: Download and Configure Emscripten SDK
+      uses: mymindstorm/setup-emsdk@v9
+
+    - name: Configure CMake
+      if: success()
+      shell: bash
+      run: |
+        cd $GITHUB_WORKSPACE/BuildTools/Scripts/github_actions
+        ./configure_cmake.sh "emscripten" "${{runner.workspace}}" ${{ matrix.config }} ${{ matrix.toolset }}
+
+    - name: Build
+      if: success()
+      working-directory: ${{runner.workspace}}/build
+      shell: bash
+      run: cmake --build .
+
+  build-linux:
+    strategy:
+      matrix:
+        config: [Debug, Release]
+
+    runs-on: ubuntu-latest
+    name: Emscripten -> Linux, ${{ matrix.config }}
+
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Format validation
+        shell: bash
+        run: |
+          cd $GITHUB_WORKSPACE/BuildTools/FormatValidation
+          ./validate_format_linux.sh
+
+      - name: Download Ninja
+        uses: actions/checkout@master
+
+      - name: Configure Ninja
+        uses: seanmiddleditch/gha-setup-ninja@master
+
+      - name: Download and Configure Emscripten SDK
+        uses: mymindstorm/setup-emsdk@v9
+
+      - name: Configure CMake
+        if: success()
+        shell: bash
+        run: |
+          cd $GITHUB_WORKSPACE/BuildTools/Scripts/github_actions
+          chmod +x configure_cmake.sh
+          ./configure_cmake.sh "emscripten" "${{runner.workspace}}" ${{ matrix.config }} ${{ matrix.toolset }}
+
+      - name: Build
+        if: success()
+        working-directory: ${{runner.workspace}}/build
+        shell: bash
+        run: cmake --build .  
```

Comment:
This will build only static libraries and does not check for linking errors.
You should add simple project to `BuildTools/Emscripten`.